### PR TITLE
doc: assumeutxo prune and index notes

### PR DESCRIPTION
**Pull request by Sjors**
Based on recent comments on #27596.

### CODE REVIEW

File: doc/design/assumeutxo.md [hunk 1]
```diff
@@ -3,8 +3,44 @@
...
+directories, each multiple gigabytes in size (growing to roughly the size of
+the downloaded snapshot).
+
+### Indexes
+
+Indexes work but don't take advantage of this feature. They always start building
+from the genesis block. Once the background validation reaches the snapshot block,
+indexes will continue to build all the way to the tip.
+
+For indexes that support pruning, not that no pruning will take place between
```
Comment (flack) [inline doc/design/assumeutxo.md:32]:
```suggestion
For indexes that support pruning, note that no pruning will take place between
```

File: doc/design/assumeutxo.md [hunk 2]
```diff
@@ -3,8 +3,44 @@
...
+the snapshot and the tip, until the background sync has completed - after which
+everything is pruned. Depending on how old the snapshot is, this may temporarily
+use a significant amount of disk space.
+
+## Generating a snapshot
+
+The RPC command `dumptxoutset` can be used to generate a snapshot. This can be used
+on one node to create a snapshot on one node that you wish to load on another node.
```
Comment (flack) [inline doc/design/assumeutxo.md:40]:
```suggestion
to create a snapshot on one node that you wish to load on another node.
```

File: doc/design/assumeutxo.md [hunk 3]
```diff
@@ -3,8 +3,44 @@
...
+
+### Pruning
+
+A pruned node can load a snapshot. To save space, it's possible to delete the
+snapshot `.dat` file as soon as `loadtxoutset` finishes.
+
+The minimum `-dbcache` setting is 550 MiB, but this functionality ignores that
+minimum and uses at least 1100 MiB.
+
+As the background sync continues there will temporarily be two chainstate
```
Comment (pablomartin4btc) [inline doc/design/assumeutxo.md:22]:
```suggestion
As the background sync continues there will be temporarily two chainstate
```

**Review by pablomartin4btc — COMMENTED** [review]
ACK

File: doc/design/assumeutxo.md [hunk 4]
```diff
@@ -3,8 +3,44 @@
...
+There is currently no canonical source for snapshots, but any downloaded snapshot
+will be checked against a hash that's been hardcoded in source code.
+
+Once you've obtained the snapshot, you can use the RPC command `loadtxoutset` to
+load it.
+
+### Pruning
+
+A pruned node can load a snapshot. To save space, it's possible to delete the
+snapshot `.dat` file as soon as `loadtxoutset` finishes.
```
Comment (jamesob) [inline doc/design/assumeutxo.md:17]: I don't think snapshot files necessarily have `.dat` extensions.

File: doc/design/assumeutxo.md [hunk 5]
```diff
@@ -3,8 +3,44 @@
...
+
+The minimum `-dbcache` setting is 550 MiB, but this functionality ignores that
+minimum and uses at least 1100 MiB.
+
+As the background sync continues there will temporarily be two chainstate
+directories, each multiple gigabytes in size (growing to roughly the size of
+the downloaded snapshot).
```
Comment (jamesob) [inline doc/design/assumeutxo.md:24]: Almost certainly exceeding the size of the serialized snapshot.

**Review by jamesob — COMMENTED** [review]
ACK modulo outstanding comments

**Comment by Sjors** [comment]
Addressed the above comments.

**Review by pablomartin4btc — APPROVED** [review]
re ACK 03f82087f6ce1c29327f34d12945200494e6956d

File: doc/design/assumeutxo.md [hunk 6]
```diff
@@ -3,8 +3,44 @@
...
+
+### Indexes
+
+Indexes work but don't take advantage of this feature. They always start building
+from the genesis block. Once the background validation reaches the snapshot block,
+indexes will continue to build all the way to the tip.
+
+For indexes that support pruning, note that no pruning will take place between
+the snapshot and the tip, until the background sync has completed - after which
```
Comment (vostrnad) [inline doc/design/assumeutxo.md:33]:
```suggestion
the snapshot and the tip until the background sync has completed, after which
```

**Review by ryanofsky — APPROVED** [review]
ACK 03f82087f6ce1c29327f34d12945200494e6956d. Nice changes, these seem like very helpful notes

**Pull request merged**